eng/scripts/Language-Settings.ps1

```diff
@@ -0,0 +1,116 @@
+$Language = "dotnet"
+$PackageRepository = "Nuget"
+$packagePattern = "*.nupkg"
+$MetadataUri = "https://raw.githubusercontent.com/Azure/azure-sdk/master/_data/releases/latest/dotnet-packages.csv"
+
+function Extract-dotnet-PkgProperties ($pkgPath, $serviceName, $pkgName)
+{
+  $projectPath = Join-Path $pkgPath "src" "$pkgName.csproj"
+  if (Test-Path $projectPath)
+  {
+    $projectData = New-Object -TypeName XML
+    $projectData.load($projectPath)
+    $pkgVersion = Select-XML -Xml $projectData -XPath '/Project/PropertyGroup/Version'
+    return [PackageProps]::new($pkgName, $pkgVersion, $pkgPath, $serviceName)
+  }
+  else
+  {
+    return $null
+  }
+}
+
+# Returns the nuget publish status of a package id and version.
+function IsNugetPackageVersionPublished($pkgId, $pkgVersion) 
+{
+  $nugetUri = "https://api.nuget.org/v3-flatcontainer/$($pkgId.ToLowerInvariant())/index.json"
+
+  try
+  {
+    $nugetVersions = Invoke-RestMethod -MaximumRetryCount 3 -uri $nugetUri -Method "GET"
+    return $nugetVersions.versions.Contains($pkgVersion)
+  }
+  catch
+  {
+    $statusCode = $_.Exception.Response.StatusCode.value__
+    $statusDescription = $_.Exception.Response.StatusDescription
+
+    # if this is 404ing, then this pkg has never been published before
+    if ($statusCode -eq 404) {
+      return $False
+    }
+
+    Write-Host "Nuget Invocation failed:"
+    Write-Host "StatusCode:" $statusCode
+    Write-Host "StatusDescription:" $statusDescription
+    exit(1)
```

Comment:
For now I want to keep these functions unchanged but we should do a pass to ensure none of these shared functions call exit directly. That is usually not a great thing as you don't know the outer context of what the script is trying to do and it isn't great to have the script exit in all cases. We should instead return failure and let the caller decide what to do.